ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -95,7 +101,31 @@ public HelixParticipant(ClusterMapConfig clusterMapConfig, HelixFactory helixFac
     if (clusterName.isEmpty()) {
       throw new IllegalStateException("Cluster name is empty in clusterMapConfig");
     }
-    manager = helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.PARTICIPANT, zkConnectStr);
+
+    // JING TODO:
+    // 1. How to differentiate it's a on-prem or azure participant? How to differentiate the frontend and the server?
+    // 2. Is the following operation safe for On-Prem participant?
+
+    // Cloud Event Callback Property: A property to define the maintenance event handling behavior for a Helix manager.
+    // User can specify which callback implementation class they want to use for Helix operations, if not specified, it will use org.apache.helix.cloud.event.helix.DefaultCloudEventCallbackImpl.class by default
+    CloudEventCallbackProperty callbackProperty = new CloudEventCallbackProperty(Collections.emptyMap());
+
+    // Enable/disable Helix optional operations for a Helix manager via CloudEventCallbackProperty.
+    callbackProperty.setHelixOperationEnabled(HelixOperation.ENABLE_DISABLE_INSTANCE, true);
+    callbackProperty.setHelixOperationEnabled(HelixOperation.MAINTENANCE_MODE, true);
+
+    // Helix Cloud Property: General cloud-related property for a Helix manager, includes CloudEventCallbackProperty.
+    HelixCloudProperty cloudProperty = new HelixCloudProperty(new CloudConfig(new ZNRecord(clusterName)));
```

Comment:
As we discussed offline, there are two ways to supply a cloud config, you could build one in code, or read from zk in ZKHelixManager constructor. `new CloudConfig(new ZNRecord(clusterName)` this means a default cloud config which is usually used when cloud is not enabled. When on cloud, if the cloud config is to be read from zk, then it's fine; but if a cloud config is meant to be built in code, then you will need to do more here.